repo_manager/github/variables.py

```diff
@@ -0,0 +1,227 @@
+import json
+from typing import Any
+
+from actions_toolkit import core as actions_toolkit
+
+from github.Repository import Repository
+
+from repo_manager.schemas.secret import Secret
+
+
+def diff_option(key: str, expected: Any, repo_value: Any) -> str | None:
+    if expected is not None:
+        if expected != repo_value:
+            return f"{key} -- Expected: {expected} Found: {repo_value}"
+    return None
+
+
+def _get_repo_variables(repo: Repository, path: str = "actions") -> Any:
+    status, headers, raw_data = repo._requester.requestJson(
+        "GET", f"{repo.url}/{path}/variables"
+    )
+    if status != 200:
+        raise Exception(
+            f"Unable to get repo's variables. Status: {status}. Error: {json.loads(raw_data)['message']}"
+        )
+    try:
+        variable_data = json.loads(raw_data)
+    except json.JSONDecodeError as exc:
+        raise Exception(f"Github apu returned invalid json {exc}")
+
+    return variable_data["variables"]
+
+
+def _get_repo_variable_names(repo: Repository, path: str = "actions") -> set[str]:
+    return {variable["name"] for variable in _get_repo_variables(repo, path)}
+
+
+def _get_repo_variable_dict(repo: Repository, path: str = "actions") -> dict[str, Any]:
+    return {variable["name"]: variable for variable in _get_repo_variables(repo, path)}
+
+
+def create_variable(
```

Comment:
pygithub has support for managing variables, can that be used instead of re-implementing the API calls in the action?